onnxruntime/core/graph/graph_utils.cc

```diff
@@ -422,6 +422,22 @@ size_t RemoveNodeOutputEdges(Graph& graph, Node& node) {
   return output_edges.size();
 }
 
+void BuildTensorProtoForInitializer(const Tensor& out_tensor, const NodeArg& node_arg,
+                                    ONNX_NAMESPACE::TensorProto& tensor_proto) {
```

Comment:
>ONNX_NAMESPACE::TensorProto& tensor_proto [](start = 36, length = 41)

I'm make this the return type to guarantee you get a clean instance to start with. RVO will ensure there's no unnecessary copy.